ddp-workspace/projects/ddp-sdk/src/lib/interceptors/auth-interceptor.service.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { HttpInterceptor, HttpRequest, HttpHandler, HttpEvent } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { Auth0AdapterService } from '../services/authentication/auth0Adapter.service';
+import { SessionMementoService } from '../services/sessionMemento.service';
+
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+    constructor(
+        private session: SessionMementoService,
+        private auth0: Auth0AdapterService) { }
+
+    public intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+        if (this.session.isSessionExpired()) {
+            this.auth0.logout('session-expired');
+        } else {
+            return next.handle(req);
+        }
+    }
+}
```

Comment:
Angular HttpInterceptors allow intercepting all HTTP requests, so we can check that the session is alive and prevent sending invalid requests. Unauthenticated requests will work as before.

Reply:
Will this only pick up on traffic back to pepper backend?  Or would it also run for calls to other URLs?

Reply:
This will pick up on traffic only back to pepper backend.